adafruit_esp32spi/adafruit_esp32spi.py

```diff
@@ -779,6 +786,37 @@ def set_analog_write(self, pin, analog_value):
         if resp[0][0] != 1:
             raise RuntimeError("Failed to write to pin")
 
+    def set_digital_read(self, pin):
+        """
+        Get the digital input value of pin. Returns the boolean value of the pin.
+
+        :param int pin: ESP32 GPIO pin to read from.
+        """
+        resp = self._send_command_get_response(_SET_DIGITAL_READ_CMD,
+                                               ((pin,),))[0]
+        if resp[0] == 0:
+            return False
+        elif resp[0] == 1:
+            return True
+        else:
+            raise ValueError("_SET_DIGITAL_READ response error", resp[0])
+
+    def set_analog_read(self, pin, atten=ADC_ATTEN_DB_11):
+        """
+        Get the analog input value of pin. Returns an int between 0 and 65536.
+
+        :param int pin: ESP32 GPIO pin to read from.
+        :param int atten: attenuation constant
+        """
+        resp = self._send_command_get_response(_SET_ANALOG_READ_CMD,
+                                               ((pin,), (atten,)))
+        resp_analog = struct.unpack('<i', resp[0])
+        if resp_analog[0] < 0:
+            raise ValueError("_SET_ANALOG_READ parameter error", resp_analog[0])
```

Comment:
Could you be more verbose with this error? If not that's OK.

Reply:
Done. Yes, currently this should only happen if the pin number doesn't correspond to an ADC1 channel.